lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2241,6 +2270,27 @@ void TFDeabstraction::formGraphOp(SILTensorOpInfo &opInfo,
       }
       break;
     }
+    case SILTensorOpInfo::OperandClass::ExtractingDTypeList: {
```

Comment:
Random question, do we need $extractingDTypeList at all?  Why not fold this into the Normal case for types?

If this is necessary, then +1 for me.  This is a very nice approach.

Reply:
The normal handling of metatypes I mean.

Reply:
If it had no suffix, it'd be unclear whether the metatype should be deabstracted to _one_ metatype or _a list_ of metatypes.

Reply:
Oh right, sigh.  :-)  Can we use $array on metatypes to represent this?

Reply:
SGTM. @njjiang Could you drop `$extractingDTypeList` overload `$array`?

@lattner There's another hacky thing we should implement to get the full Dataset API. That's telling `#tfop` to turn an TF value aggregate to a list of unknown shapes (`nil`) for each tensor element. Not sure what that suffix should be called.

Reply:
Yes. `$unknownShapeList` WFM.